### DISCUSSION TITLE
Active-Passive Milvus Deployments

### DISCUSSION THREAD

**Discussion by ramish-sidiqui** [milvus-io/milvus#35678] [Q&A and General discussion]
Hi guys,

I am trying to implement an active passive milvus deployment architecture, what that essentially means is that if one of the milvus deployment becomes unhealthy for some reason, passive milvus deployment will be used instead. Milvus CDC is implemented between the two milvus deployments so they are in sync. For more understanding you can see the diagram below for clarity.
![image](https://github.com/user-attachments/assets/fa680247-a07f-46be-b86b-164cf6c66ad6)

Now here is my question, what should be the criteria against which i can instruct the ALB through my Lambda function to switch the deployment. I have read several threads
https://github.com/milvus-io/milvus/discussions/35016#discussioncomment-10165547
https://github.com/milvus-io/milvus/discussions/34459#discussion-6901219
These specify the each node/pod has health checks available which are exposed on port 9091/healthz, but these health checks are for the node/pod itself. How can i judge that my milvus deployment is in error state overall. So i can switch to passive deployment.

**Comment by xiaofan-luan** [comment]
We are working on a better checkhealth 

https://github.com/milvus-io/milvus/issues/34264

I think the automatic switch relies on two things:
1. Milvus, K8s and network is down and you should easily detect it.
2. check health fail (this may or may not critical)

Let's probably have a discussion about it if necessary, my email james.luan@zilliz.com

**Reply by ramish-sidiqui** [reply]
Good to hear that this is already on the roadmap, any idea with what milvus version this feature might be released ?

Regarding your second point, will the new check health functionality provide if the error is non-critical and can be recovered from or critical non-recoverable, in this case switch to passive milvus is required.

**Reply by xiaofan-luan** [reply]
We have multiple different strategy. But I believe whether it should switch or not should be defined by user.

**Reply by ramish-sidiqui** [reply]
On which milvus release this feature will be included ?

**Reply by xiaofan-luan** [reply]
> On which milvus release this feature will be included ?

We offered cdc already but the active standby is not offered as opensource.

This will be an enterprise feature gonna to release on zilliz cloud by the end of september because the traffic switch and health check requires product design.
